### PR TITLE
Closes #6354: Tests for RUCSS API & CRON Refactor

### LINKED ISSUE #6354: Add tests for RUCSS CRON Refactor
Add automated tests for RUCSS API & CRON Refactor.

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description
Adds tests for RUCSS API & CRON Refactor

Fixes #6354

### CODE REVIEW

File: tests/Fixtures/inc/Engine/Common/JobManager/APIHandler/.DS_Store
Comment (Khadreal) [inline tests/Fixtures/inc/Engine/Common/JobManager/APIHandler/.DS_Store:1]: I think this should be in gitignore file

Comment (jeawhanlee) [reply tests/Fixtures/inc/Engine/Common/JobManager/APIHandler/.DS_Store:1]: it shouldn't even be there in the first place, Thanks @Khadreal

**Review by Khadreal — APPROVED** [review]
Looks good to me

**Review by wordpressfan — CHANGES_REQUESTED** [review]
We have failed tests, mainly because this method:
```
WP_Rocket\Engine\Optimization\RUCSS\Admin\Settings::add_localize_script_data
```

seems like it is moved to

https://github.com/wp-media/wp-rocket/blob/0b20b4047605be6f190d5ba9aa9b18f8bf20222d/inc/Engine/Saas/Admin/Subscriber.php#L160-L162

**Comment by jeawhanlee** [comment]
> We have failed tests, mainly because this method:
> 
> ```
> WP_Rocket\Engine\Optimization\RUCSS\Admin\Settings::add_localize_script_data
> ```
> 
> seems like it is moved to
> 
> https://github.com/wp-media/wp-rocket/blob/0b20b4047605be6f190d5ba9aa9b18f8bf20222d/inc/Engine/Saas/Admin/Subscriber.php#L160-L162

I saw it but did not touch this yet as it's not related to the refactored RUCSS CRON & API Module plus I think they are related to a sub-task of 3.16 that @Tabrisrp worked on, so I don't have enough context on how they will be implemented.

**Review by remyperona — APPROVED** [review]

**Pull request merged**